recipes/dcmtk/all/conanfile.py

```diff
@@ -167,147 +170,116 @@ def package(self):
-        # dcmrt
-        self.cpp_info.components["dcmrt"].includedirs.append(includedir)
-        self.cpp_info.components["dcmrt"].libs = ["dcmrt"]
-        self.cpp_info.components["dcmrt"].requires = ["ofstd", "oflog", "dcmdata", "dcmimgle"]
-        # dcmiod
-        self.cpp_info.components["dcmiod"].includedirs.append(includedir)
-        self.cpp_info.components["dcmiod"].libs = ["dcmiod"]
-        self.cpp_info.components["dcmiod"].requires = ["dcmdata", "ofstd", "oflog"]
-        # dcmfg
-        self.cpp_info.components["dcmfg"].includedirs.append(includedir)
-        self.cpp_info.components["dcmfg"].libs = ["dcmfg"]
-        self.cpp_info.components["dcmfg"].requires = ["dcmiod", "dcmdata", "ofstd", "oflog"]
-        # dcmseg
-        self.cpp_info.components["dcmseg"].includedirs.append(includedir)
-        self.cpp_info.components["dcmseg"].libs = ["dcmseg"]
-        self.cpp_info.components["dcmseg"].requires = ["dcmfg", "dcmiod", "dcmdata", "ofstd", "oflog"]
-        # dcmtract
-        self.cpp_info.components["dcmtract"].includedirs.append(includedir)
-        self.cpp_info.components["dcmtract"].libs = ["dcmtract"]
-        self.cpp_info.components["dcmtract"].requires = ["dcmiod", "dcmdata", "ofstd", "oflog"]
-        # dcmpmap
-        self.cpp_info.components["dcmpmap"].includedirs.append(includedir)
-        self.cpp_info.components["dcmpmap"].libs = ["dcmpmap"]
-        self.cpp_info.components["dcmpmap"].requires = ["dcmfg", "dcmiod", "dcmdata", "ofstd", "oflog"]
+
+        def register_components(components):
+            for target_lib, requires in components.items():
+                self.cpp_info.components[target_lib].names["cmake_find_package"] = target_lib
+                self.cpp_info.components[target_lib].names["cmake_find_package_multi"] = target_lib
+                self.cpp_info.components[target_lib].builddirs.append(self._module_subfolder)
+                self.cpp_info.components[target_lib].build_modules["cmake_find_package"] = [self._module_file_rel_path]
+                self.cpp_info.components[target_lib].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
+                self.cpp_info.components[target_lib].libs = [target_lib]
+                self.cpp_info.components[target_lib].includedirs.append(os.path.join("include", "dcmtk"))
+                self.cpp_info.components[target_lib].requires = requires
+
+                if target_lib == "ofstd":
+                    if self.settings.os == "Windows":
+                        self.cpp_info.components[target_lib].system_libs.extend([
+                            "iphlpapi", "ws2_32", "netapi32", "wsock32"
+                        ])
+                    elif self.settings.os == "Linux":
+                        self.cpp_info.components[target_lib].system_libs.append("m")
+                        if self.options.with_multithreading:
+                            self.cpp_info.components[target_lib].system_libs.append("pthread")
```

Comment:
Could this be moved to the package_info scope? seems odd to cover over this for every lib

Reply:
moved outside the loop